docs/control-structures.rst

```diff
@@ -117,14 +143,14 @@ External Function Calls
 The expressions ``this.g(8);`` and ``c.g(2);`` (where ``c`` is a contract
 instance) are also valid function calls, but this time, the function
 will be called "externally", via a message call and not directly via jumps.
-Please note that function calls on ``this`` cannot be used in the constructor, as the
-actual contract has not been created yet.
+
+...note::
+     You cannot use function calls on ``this`` in the constructor, as the actual contract has not been created yet.
```

Comment:
While this hasn't been introduced in this PR, but there is one use case for `this`: `address(this)` can be used to retrieve the address of the contract, even in the constructor. What is discouraged is doing calls on `this`, because that would call into a not-yet existing contract. it may be worth expanding this text.

Reply:
Took a go at adding this.